### PR TITLE
feat: support sign with password & user info operation

### PR DESCRIPTION
**Pull request by QiShaoXuan**
Still some detail need implement: https://github.com/toeverything/AFFiNE/issues/3556

### CODE REVIEW

File: apps/server/src/modules/auth/guard.ts [hunk 1]
```diff
@@ -92,8 +92,10 @@ class AuthGuard implements CanActivate {
         return false;
       }
 
-      // @ts-expect-error body is user here
-      req.user = body.user;
+      req.user = await this.prisma.user.findUnique({
```
Comment (Brooooooklyn) [inline apps/server/src/modules/auth/guard.ts:95]: duplicate database query here

File: apps/server/src/modules/auth/next-auth-options.ts [hunk 2]
```diff
@@ -217,10 +219,17 @@ export const NextAuthOptionsProvider: FactoryProvider<NextAuthOptions> = {
             session.user.id = user.id;
             // @ts-expect-error Third part library type mismatch
             session.user.image = user.image ?? user.avatarUrl;
+            // @ts-expect-error Third part library type mismatch
+            session.user.hasPassword = Boolean(user.password);
           } else {
             // technically the sub should be the same as id
             // @ts-expect-error Third part library type mismatch
             session.user.id = token.sub;
+            const databaseUser = await prisma.user.findUnique({
```
Comment (Brooooooklyn) [inline apps/server/src/modules/auth/next-auth-options.ts:228]: duplicate database query here

Comment (Brooooooklyn) [reply apps/server/src/modules/auth/next-auth-options.ts:228]: `user` in this function params should be the same with the `prisma.user.findUnique`

File: apps/server/src/modules/auth/guard.ts (hunk 1 above)
Comment (darkskygit) [reply apps/server/src/modules/auth/guard.ts:95]: user session was filter by next-auth, we dont has enough information

File: packages/component/src/components/auth-components/change-email-page.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+type User = {
+  id: string;
+  name: string;
+  email: string;
+  image: string;
+};
+function validateEmail(email: string) {
+  return new RegExp(
+    /^(?:(?:[^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@(?:(?:\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|((?:[a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+  ).test(email);
```
Comment (darkskygit) [inline packages/component/src/components/auth-components/change-email-page.tsx:17]: extract regex init as a top level const

**Review by himself65 — APPROVED** [review]
In the last meeting, we saw some problem that if you have 2 tabs and you logout the account in the second one, the first tab will crash if you are opening account profile.

Please fix this issue

**Comment by QiShaoXuan** [comment]
> In the last meeting, we saw some problem that if you have 2 tabs and you logout the account in the second one, the first tab will crash if you are opening account profile.
> 
> Please fix this issue

Fixed in https://github.com/toeverything/AFFiNE/pull/3504/commits/9768d2c4f92c0e42d998c687dc0b9ce231ef3ea9

**Pull request merged**